src/checkpoint/checkpoint_txn.c

```diff
@@ -1567,6 +1567,7 @@ __wt_checkpoint_db(WT_SESSION_IMPL *session, const char *cfg[], bool waiting)
     uint32_t orig_flags;
     bool checkpoint_cleanup, flush, flush_sync;
 
+    WT_STAT_CONN_SET(session, checkpoint_state, WTI_CHECKPOINT_STATE_ACTIVE);
```

Comment:
The most important change is this one.
@korteland, do you remember our conversation about checkpoint states? Did we have a reason for setting this state at the start of the API call instead of inside the actual checkpoint internal API?

Reply:
AFAICT the original choice only set the ACTIVE state when checkpoint was called by the user, but that state was missed when called internally by WT. One possibility was that setting it there was near setting the other statistics. IMO it seems like this change is better because it reflects any checkpoint being ACTIVE no matter how it was generated. That seems more accurate to me.

Reply:
Sorry I missed the ping @etienneptl, I've had problems with not receiving Github emails recently.

It's inadvertent that the original location didn't catch checkpoint being called internally, I agree this location is better, thanks for catching it.

Reply:
Thank you, @sueloverso and @korteland! I will proceed with the merge.